SDK/MobileCenter/Microsoft.Azure.Mobile.UWP/Utils/ApplicationLifecycleHelper.cs

```diff
@@ -46,17 +45,23 @@ public ApplicationLifecycleHelper()
                 // the start
                 _started = true;
             }
-            try
+
+            // Subscribe to unhandled errors events
+            CoreApplication.UnhandledErrorDetected += (sender, eventArgs) =>
             {
-                Application.Current.UnhandledException += (sender, eventArgs) =>
+                try
                 {
-                    UnhandledExceptionOccurred?.Invoke(sender, new UnhandledExceptionOccurredEventArgs(eventArgs.Exception));
-                };
-            }
-            catch (Exception exception)
-            {
-                MobileCenterLog.Warn(MobileCenterLog.LogTag, "Cannot subscribe to unhandled exceptions from XAML application", exception);
-            }
+                    // intentionally propagating exception to get the exception object that crashed the app.
+                    eventArgs.UnhandledError.Propagate();
+                }
+                catch (Exception exception)
+                {
+                    UnhandledExceptionOccurred?.Invoke(sender, new UnhandledExceptionOccurredEventArgs(exception));
+
+                    // If we don't throw exception - app will not be crashed. We need to throw to not change the app behavior.
+                    throw;
```

Comment:
> Since UnhandledError.Propagate marks the error as Handled, rethrow in order to only Log and not Handle
See (C++ but event the same): https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.core.icoreapplicationunhandlederror#events_

Also in example above `Handled` is checked, so if we don't check this we will log handled errors as well.
Updated comment for that to make it more clear.